lib/srv/desktop/windows_server.go

```diff
@@ -1251,6 +1257,49 @@ func (s *WindowsService) generateCredentials(ctx context.Context, username, doma
 	return certDER, keyDER, nil
 }
 
+// trackSession creates a session tracker for the given sessionID and
+// attributes, and starts a goroutine to continually extend the tracker
+// expiration while the session is active. Once the given ctx is closed,
+// the tracker will be marked as terminated.
+func (s *WindowsService) trackSession(ctx context.Context, id *tlsca.Identity, windowsUser string, sessionID string, desktop types.WindowsDesktop) error {
+	s.cfg.Log.Debugf("Starting tracker for session %v", sessionID)
+	initiator := &types.Participant{
+		ID:   id.Username,
+		User: id.Username,
+	}
+
+	tracker, err := types.NewSessionTracker(types.SessionTrackerSpecV1{
+		SessionID:    sessionID,
+		Kind:         string(types.WindowsDesktopSessionKind),
+		State:        types.SessionState_SessionStateRunning,
+		Hostname:     s.cfg.Hostname,
+		Address:      desktop.GetAddr(),
+		ClusterName:  s.clusterName,
+		Login:        windowsUser,
+		Participants: []types.Participant{*initiator},
+		HostUser:     initiator.User,
+		Created:      s.cfg.Clock.Now(),
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	_, err = s.cfg.AuthClient.CreateSessionTracker(ctx, tracker)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	go func() {
+		ticker := s.cfg.Clock.NewTicker(defaults.SessionTrackerExpirationUpdateInterval)
+		defer ticker.Stop()
+		if err := services.TrackSession(s.closeCtx, s.cfg.AuthClient, tracker, ticker, ctx.Done()); err != nil {
+			s.cfg.Log.Debugf("Error tracking session state for session %v", sessionID)
```

Comment:
Should you include the error in the log line?

Reply:
```suggestion
			s.cfg.Log.WithError(err).Debugf("Error tracking session state for session %v", sessionID)
```